File: im-mobile/lib/models/batch_operation_request.dart
// 批量操作请求模型
import 'batch_operation_type.dart';
import 'package:json_annotation/json_annotation.dart';

part 'batch_operation_request.g.dart';

@JsonSerializable()
class BatchOperationRequest {
  final List<String> messageIds;
  final BatchOperationType operationType;
  final String? targetConversationId;
  final List<String>? targetUserIds;
  final bool? keepOriginal;
  final Map<String, dynamic>? additionalParams;
  final String? reason;
  final bool? deleteAfterMove;
  final bool? asyncExecution;
  final DateTime? scheduledTime;

  BatchOperationRequest({
    required this.messageIds,
    required this.operationType,
    this.targetConversationId,
    this.targetUserIds,
    this.keepOriginal,
    this.additionalParams,
    this.reason,
    this.deleteAfterMove,
    this.asyncExecution,
    this.scheduledTime,
  });

  factory BatchOperationRequest.fromJson(Map<String, dynamic> json) =>
      _$BatchOperationRequestFromJson(json);

  Map<String, dynamic> toJson() => _$BatchOperationRequestToJson(this);

  BatchOperationRequest copyWith({
    List<String>? messageIds,
    BatchOperationType? operationType,
    String? targetConversationId,
    List<String>? targetUserIds,
    bool? keepOriginal,
    Map<String, dynamic>? additionalParams,
    String? reason,
    bool? deleteAfterMove,
    bool? asyncExecution,
    DateTime? scheduledTime,
  }) {
    return BatchOperationRequest(
      messageIds: messageIds ?? this.messageIds,
      operationType: operationType ?? this.operationType,
      targetConversationId: targetConversationId ?? this.targetConversationId,
      targetUserIds: targetUserIds ?? this.targetUserIds,
      keepOriginal: keepOriginal ?? this.keepOriginal,
      additionalParams: additionalParams ?? this.additionalParams,
      reason: reason ?? this.reason,
      deleteAfterMove: deleteAfterMove ?? this.deleteAfterMove,
      asyncExecution: asyncExecution ?? this.asyncExecution,
      scheduledTime: scheduledTime ?? this.scheduledTime,
    );
  }
}

File: im-desktop/types/batch-operation-request.ts
/**
 * 批量操作请求类型
 */
export interface BatchMessageOperationRequest {
  messageIds: string[];
  operationType: BatchOperationType;
  targetConversationId?: string;
  targetUserIds?: string[];
  keepOriginal?: boolean;
  additionalParams?: Record<string, any>;
  reason?: string;
  deleteAfterMove?: boolean;
  asyncExecution?: boolean;
  scheduledTime?: string;
}

/**
 * 批量操作结果类型
 */
export interface BatchOperationResult {
  batchId: string;
  operationType: BatchOperationType;
  totalCount: number;
  successCount: number;
  failureCount: number;
  skippedCount: number;
  successMessageIds: string[];
  failures: FailedOperation[];
  skipped: SkippedOperation[];
  startTime: string;
  endTime: string;
  durationMs: number;
  status: BatchOperationStatus;
  asyncExecution: boolean;
  asyncTaskId?: string;
  operatorId: string;
  operatorName?: string;
  targetConversationId?: string;
  extraData?: Record<string, any>;
  generatedMessageIds?: string[];
}

export interface FailedOperation {
  messageId: string;
  errorCode: string;
  errorMessage: string;
  reason: FailureReason;
}

export interface SkippedOperation {
  messageId: string;
  reason: SkipReason;
  description: string;
}

export enum BatchOperationStatus {
  PENDING = 'PENDING',
  RUNNING = 'RUNNING',
  COMPLETED = 'COMPLETED',
  PARTIAL_SUCCESS = 'PARTIAL_SUCCESS',
  FAILED = 'FAILED',
  CANCELLED = 'CANCELLED',
}

export enum FailureReason {
  PERMISSION_DENIED = 'PERMISSION_DENIED',
  MESSAGE_NOT_FOUND = 'MESSAGE_NOT_FOUND',
  ALREADY_DELETED = 'ALREADY_DELETED',
  RECALL_TIMEOUT = 'RECALL_TIMEOUT',
  NETWORK_ERROR = 'NETWORK_ERROR',
  SERVER_ERROR = 'SERVER_ERROR',
  VALIDATION_ERROR = 'VALIDATION_ERROR',
}

export enum SkipReason {
  ALREADY_PROCESSED = 'ALREADY_PROCESSED',
  NO_PERMISSION = 'NO_PERMISSION',
  FILTERED_OUT = 'FILTERED_OUT',
  DUPLICATE_REQUEST = 'DUPLICATE_REQUEST',
}

/**
 * 批量操作历史查询参数
 */
export interface BatchOperationHistoryQuery {
  page?: number;
  size?: number;
  status?: BatchOperationStatus;
  operationType?: BatchOperationType;
  startTime?: string;
  endTime?: string;
}

/**
 * 批量操作任务
 */
export interface BatchOperationTask {
  id: string;
  operatorId: string;
  operationType: BatchOperationType;
  totalCount: number;
  successCount: number;
  failureCount: number;
  skippedCount: number;
  messageIds: string[];
  status: BatchOperationStatus;
  progress: number;
  createdAt: string;
  updatedAt: string;
  completedAt?: string;
}

import { BatchOperationType } from './batch-operation';
